src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,45 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
+    private static final int MAX_SIZE = 100;
+    private int size;
+    private T[] stack;
+
+    public MateStack() {
+        stack = (T[]) new Object[MAX_SIZE];
+        size = 0;
+    }
+
     public void push(T value) {
+        if (size >= MAX_SIZE - 1) {
+            throw new StackOverflowError();
```

Comment:
let's throw a RuntimeException with an informative message

we should not throw Errors

Reply:
Why do we need to throw  RunTimeException if java already has an error that according to this situation? (It's only for my information)